sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java

```diff
@@ -123,31 +135,15 @@ public void stop() throws ConnectException {
   @Override
   public Map<TopicPartition, OffsetAndMetadata> preCommit(
       Map<TopicPartition, OffsetAndMetadata> currentOffsets) {
-    Map<TopicPartition, OffsetAndMetadata> result = new HashMap<>();
-
-    log.debug("{} performing preCommit with offsets: {}", name(), currentOffsets);
-
-    currentOffsets.entrySet().stream()
-        .filter(
-            e ->
-                e.getValue() != null
-                    && partitions.get(e.getKey()) != null
-                    && partitions.get(e.getKey()).shouldFlush())
-        .forEach(
-            e -> {
-              BlockGZIPFileWriter blockWriter = partitions.get(e.getKey()).getWriter();
-              log.debug(
-                  "{} preparing offsets for partition: {}, totalCompressedSize: {}",
-                  name(),
-                  e.getKey(),
-                  blockWriter.getTotalCompressedSize());
-              result.put(e.getKey(), e.getValue());
-            });
 
-    if (!result.isEmpty()) {
-      flush(result);
-    }
+    partitions.values().stream()
+        .filter(PartitionWriter::shouldFlush)
```

Comment:
It makes sense to leave some comments explaining which cases are covered by `shouldFlush` or `shouldFlushBefore` calls. For example here we flush if:
- no thresholds are defined (in this case flush interval is controlled by Kafka Connect settings)
- defined flush interval + grace period passed

Reply:
Added comments to all places where the flushing is done. Also added comments to all three method with the logic controlling the flushing.